### PR TITLE
Fix contract sent with reqHistoricalTicks

### PR DESCRIPTION
**Pull request by ant-fx**
Proposed fix for the issue mentioned in https://github.com/stoqey/ib/issues/86.

I'm not 100% sure what's going on with IB's contract format when calling their API, looking at their Python api they seem to send different data depending on the function call.

The update here uses the Python api as a reference and has a test. The test will only pass if you have the required market subscription unfortunately :(

```python
flds += [make_field(OUT.REQ_HISTORICAL_TICKS),
         make_field(reqId),
         make_field(contract.conId),
         make_field(contract.symbol),
         make_field(contract.secType),
         make_field(contract.lastTradeDateOrContractMonth),
         make_field(contract.strike),
         make_field(contract.right),
         make_field(contract.multiplier),
         make_field(contract.exchange),
         make_field(contract.primaryExchange),
         make_field(contract.currency),
         make_field(contract.localSymbol),
         make_field(contract.tradingClass),
         make_field(contract.includeExpired),
         make_field(startDateTime),
         make_field(endDateTime),
         make_field(numberOfTicks),
         make_field(whatToShow),
         make_field(useRth),
         make_field(ignoreSize)]

miscOptionsString = ""
if miscOptions:
    for tagValue in miscOptions:
        miscOptionsString += str(tagValue)
flds += [make_field(miscOptionsString),]
```

I didn't want to touch the api testing code too much because I believe @Tsopic is working on improvements to that here https://github.com/stoqey/ib/pull/87

Please let me know if any improvements are needed. It would be good to take a closer look at the API and see if the contract tokeniser could be shared between multiple API calls but that might not be possible.

### CODE REVIEW

**Review by mfrener — APPROVED** [review]
We use the Java client as a reference, but also there REQ_TICK_BY_TICK_DATA is not implemented by b.send(contract) (would be this.encodeContract(contract) for us), but they add send contract attributes one by one, as last two send's are actually different... https://github.com/stoqey/ib/blob/master/ref/client/EClient.java
So your fix is correct and in-line with the Java client code - thanks & approved ;)

**Comment by mfrener** [comment]
I have found one more issue on the HISTORICAL_TICKS decoder (unprocessed data left on queue).
Will do a follow-up PR containing that fix, as well adding support for historical tick data to IBApiNext (+IBApiNext.getHistoricalTicksMid +IBApiNext.getHistoricalTicksBidAsk +IBApiNext.getHistoricalTicksLast)

**Comment by mfrener** [comment]
Here it is https://github.com/stoqey/ib/pull/89

File: src/tests/unit/api/api.test.ts [hunk 1]
```diff
@@ -139,4 +139,53 @@ describe("IBApi Tests", () => {
 
     ib.reqPnLSingle(44, _account, null, _conId);
   });
+
+  it("Test request tick history", async function (done) {
+    const ib = new IBApi({
+      clientId: _clientId++,
+      host: TEST_SERVER_HOST,
+      port: TEST_SERVER_POST,
```
Comment (Tsopic) [inline src/tests/unit/api/api.test.ts:147]: Once the test env PR is merged

The only valid values are `IB_HOST` and `IB_PORT`

**Review by Tsopic — CHANGES_REQUESTED** [review]

**Comment by Tsopic** [comment]
Let us merge the other pr before, then I can change these values noted above myself to save you from the trouble ;)

**Comment by ant-fx** [comment]
@Tsopic sounds good!

**Review by Tsopic — APPROVED** [review]

**Comment by Tsopic** [comment]
@ant-fx merging this pr after tests pass

**Pull request merged**